tsconfig.json

```diff
@@ -0,0 +1,103 @@
+{
+  "compilerOptions": {
```

Comment:
Looking at this resource I'm a little bit lost on the current proposed settings.
https://www.typescriptlang.org/docs/handbook/babel-with-typescript.html#type-checking-and-dts-file-generation